src/github.com/t3chguy/riot-static/matrix-client/room.go

```diff
@@ -1,328 +0,0 @@
-// Copyright 2017 Michael Telatynski <7t3chguy@gmail.com>
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//     http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
-package matrix_client
-
-import (
-	"fmt"
-	"github.com/matrix-org/gomatrix"
-	"github.com/t3chguy/riot-static/utils"
-	"sort"
-	"strings"
-	"sync"
-	"time"
-)
-
-type RoomEventErrorEnum int
-
-const (
-	RoomEventsCouldNotFindEvent RoomEventErrorEnum = iota
-	RoomEventsUnknownError
-	RoomEventsFine
```

Comment:
This isn't idiomatic Go, and is a weak solution for communicating errors to the caller. It's a weak solution because:
 - It lacks the ability to give contextual information. For example, *which event* was unable to be found in `RoomEventsCouldNotFindEvent`?
 - It doesn't stringify sensibly for outputting to logs.
 - It doesn't play nicely with the Go ecosystem as it does not implement the `error` interface. (related somewhat to the previous point)

[Effective Go](https://golang.org/doc/effective_go.html#errors) has all the information you need here. Differentiate between errors using type assertions instead.